File: src/main/java/com/amazonaws/iot/fleetwise/vehiclesimulator/exceptions/EcsTaskManagerException.java
package com.amazonaws.iot.fleetwise.vehiclesimulator.exceptions;

import java.util.Map;
import kotlin.Metadata;
import kotlin.collections.MapsKt;
import kotlin.jvm.internal.Intrinsics;
import org.jetbrains.annotations.NotNull;

public final class EcsTaskManagerException extends Exception {

    private Map<String, String>  alreadyCreatedVehicleMap;

    public EcsTaskManagerException( String message,  Throwable cause) {
        super(message, cause);
        this.alreadyCreatedVehicleMap = MapsKt.emptyMap();
    }

    public EcsTaskManagerException( String message) {
        super(message);
        this.alreadyCreatedVehicleMap = MapsKt.emptyMap();
    }

    public EcsTaskManagerException( String message,  Map<String, String>  vehicleMap) {
        super(message);
        this.alreadyCreatedVehicleMap = MapsKt.emptyMap();
        this.alreadyCreatedVehicleMap = vehicleMap;
    }


    public final Map<String, String>  getAlreadyCreatedVehicleMap() {
        return this.alreadyCreatedVehicleMap;
    }
}

File: src/main/kotlin/com/amazonaws/iot/vehiclesimulator/exceptions/EcsTaskManagerException.kt
package com.amazonaws.iot.fleetwise.vehiclesimulator.exceptions

class EcsTaskManagerException : Exception {
    constructor(message: String, cause: Throwable) : super(message, cause)

    constructor(message: String) : super(message)

    // This map contains already created vehicle IDs and corresponding ECS Task IDs
    private var alreadyCreatedVehicleMap: Map<String, String> = mapOf()
    constructor(message: String, vehicleMap: Map<String, String>) : super(message) {
        alreadyCreatedVehicleMap = vehicleMap
    }
    fun getAlreadyCreatedVehicleMap(): Map<String, String> {
        return alreadyCreatedVehicleMap
    }
}
